testutils/inject/slam_service.go

```diff
@@ -49,3 +53,56 @@ func (slamSvc *SLAMService) DoCommand(ctx context.Context,
 	}
 	return slamSvc.DoCommandFunc(ctx, cmd)
 }
+
+// SLAMServiceClient represents a fake instance of the client the slam service uses to communicate with the underlying SLAM algorithm.
+type SLAMServiceClient struct {
+	v1.SLAMServiceClient
+	GetPositionFunc func(ctx context.Context, in *v1.GetPositionRequest, opts ...grpc.CallOption) (
+		*v1.GetPositionResponse, error)
+	GetPointCloudMapFunc func(ctx context.Context, in *v1.GetPointCloudMapRequest, opts ...grpc.CallOption) (
+		v1.SLAMService_GetPointCloudMapClient, error)
+	GetInternalStateFunc func(ctx context.Context, in *v1.GetInternalStateRequest, opts ...grpc.CallOption) (
+		v1.SLAMService_GetInternalStateClient, error)
+	DoCommandFunc func(ctx context.Context, in *commonv1.DoCommandRequest, opts ...grpc.CallOption) (
+		*commonv1.DoCommandResponse, error)
+}
+
+// GetPosition calls the injected GetPositionFunc or the real version.
+func (slamSvcClient *SLAMServiceClient) GetPosition(ctx context.Context, in *v1.GetPositionRequest, opts ...grpc.CallOption) (
+	*v1.GetPositionResponse, error,
+) {
+	if slamSvcClient.GetPositionFunc == nil {
+		return slamSvcClient.SLAMServiceClient.GetPosition(ctx, in, nil)
```

Comment:
why is this always passed a `nil` as the last arg?

Reply:
It's a workaround If I pass in opts directly I get the error:
```
[]grpc.CallOption does not implement grpc.CallOption (missing after method)
../rdk/testutils/inject/slam_service.go:107:46: cannot use opts (variable of type []grpc.CallOption) as type grpc.CallOption in argument to slamSvcClient.DoCommandFunc:
```

and If I pass in just the first opt (opts[0]) I get:
```
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0
```

To me, this implies that we are not passing in any args, and further investigation as to why opts wasn't working involves delving into the auto-gen files from the API which I chose to not pursue, but if if needed/

Reply:
what about `return slamSvcClient.SLAMServiceClient.GetPosition(ctx, in, opts...)`

Reply:
That solved it! It was my mistake. I thought the ... had to do with the type and not the variable

Reply:
Great call Nick